### ISSUE TITLE
error: Extensions are not supported with resolve procs or functions

### ISSUE THREAD

**Issue by cainlevy** [rmosolgo/graphql-ruby#2114]
Upgrading from 1.8.13 to 1.9.1 introduces this error:

```
/home/circleci/backend_app/vendor/bundle/ruby/2.6.0/gems/graphql-1.9.1/lib/graphql/schema/field.rb:281:in `extensions': Extensions are not supported with resolve procs or functions, (ArgumentError)
but Types::ClassroomType.availabilities has: #<Proc:0x0000560b18e4f5d8@/home/circleci/backend_app/app/graphql/types/base_object.rb:14 (lambda)>
So, it can't have extensions: [].
Use a method or a Schema::Resolver instead.
```

Not really sure what the error is about. Is it saying that I can't have any extensions and that `[]` is an extension? Or is it saying that I can't have any extensions and that my list of extensions is empty?

The failing field appears to be:

```
module Types
  class ClassroomType < BaseObject
    field :availabilities, [ClassroomAvailabilityType], null: false
  end

  class ClassroomAvailabilityType < BaseObject
  end
end
```

**Comment by swalkinshaw** [comment]
The error message is a bit misleading, but the issue here is that you're using a resolve proc. The extensions part is a red herring.

The key part is: `Types::ClassroomType.availabilities has: #<Proc:0x0000560b18e4f5d8@/home/circleci/backend_app/app/graphql/types/base_object.rb:14 (lambda)`

So check what `app/graphql/types/base_object.rb` is doing.

**Comment by cainlevy** [comment]
Thanks for the assist!

I didn't include the resolve proc in my simplified example. Is that deprecated or removed? I can't spot anything obviously relevant in the changelog for 1.9.0.

**Comment by swalkinshaw** [comment]
Actually sorry I was confused too 😓. It does *think* you have an extension. Resolve procs should still work unless you opted in to the new interpreter (that's the part I forgot).

So maybe this is a bug.

**Comment by swalkinshaw** [comment]
https://github.com/swalkinshaw/graphql-ruby/blob/93ba5121718f82e9e35bef707327278cbb694a4e/lib/graphql/schema/field.rb#L275-L287

Looks like this error message should use `new_extensions` instead of `extensions`. Then it should at least show what extension is trying to be added to the field.

**Comment by swalkinshaw** [comment]
Ah one more thing I remembered. The gem adds extensions by default in two situations: field is a connection or its *scoped*.

https://github.com/rmosolgo/graphql-ruby/blob/93ba5121718f82e9e35bef707327278cbb694a4e/lib/graphql/schema/field.rb#L242-L249

You may need to opt-out of those explicitly when creating a field.

**Comment by rmosolgo** [comment]
Ohhh this is definitely a gem bug. Thanks for your discussion here. 

When I moved the implementation of `scope: true` and `connection: true` into extensions, I forgot about this negative interference (and obviously didn't _test_ it 😖 ).  Everything I understand about your setup _should_ Just Work :tm: @cainlevy , so thanks again for sharing those details. I'll take a look soon and follow up here.

**Comment by cainlevy** [comment]
Thanks for the update! I'm happy to provide more detail if needed.

**Comment by rmosolgo** [comment]
It turns out the easiest approach was just to _support_ extensions with those procs, so I gave it a try in #2115 . @cainlevy , would you mind bundling that version with your app and seeing if it runs alright? I think my test changes should cover it, but I may have missed a spot. You can bundle it like: 

```ruby 
gem "graphql", github: "rmosolgo/graphql-ruby", branch: "support-field-extensions-with-resolve-procs"
```

If it looks ok to you, I'll cut a new version for rubygems!

**Comment by cainlevy** [comment]
yep, looks good! i have a passing build on that branch.

**Comment by rmosolgo** [comment]
Thanks for reporting this and trying out the patch, shipped in 1.9.2! Sorry for the hangup.
